src/libraries/System.Net.Primitives/tests/FunctionalTests/CookieTest/CookiePortTest.cs

```diff
@@ -13,65 +13,65 @@ public class CookiePortTest
         public CookiePortTest()
         {
             _cc = new CookieContainer();
-            _cookie = new Cookie("name", "value1", "/path", "localhost");
+            _cookie = new Cookie("name", "value1", "/path", "example.com");
             // use both space and comma as delimiter
             _cookie.Port = "\"80 110,1050, 1090 ,1100\"";
-            _cc.Add(new Uri("http://localhost/path"), _cookie);
+            _cc.Add(new Uri("http://example.com/path"), _cookie);
         }
 
         [Fact]
         public void Port_SetMultiplePorts_Port1Set()
         {
-            CookieCollection cookies = _cc.GetCookies(new Uri("http://localhost:80/path"));
+            CookieCollection cookies = _cc.GetCookies(new Uri("http://example.com:80/path"));
             Assert.Equal(1, cookies.Count);
         }
 
         [Fact]
         public void Port_SpaceDelimiter_PortSet()
         {
-            CookieCollection cookies = _cc.GetCookies(new Uri("http://localhost:110/path"));
+            CookieCollection cookies = _cc.GetCookies(new Uri("http://example.com:110/path"));
```

Comment:
why this needs to be changed?

Reply:
@wfurt See the PR description:

> In CookiePortTest.cs use example.com instead of localhost for the tests since Browser uses localhost as the Environment.MachineName and that changes the test values.

Reply:
The part I don't understand is why that would impact cookies. We don't do any networking here AFAIK, right?

Reply:
@wfurt we don't do networking but we still read the domain name (which on netcore on Unix generally means the machine name) here: https://github.com/dotnet/runtime/blob/ea8262910cc1c072e612c6cdd0693e27ea218b92/src/libraries/System.Net.Primitives/src/System/Net/CookieContainer.cs#L139-L145

That is later on used for this check: https://github.com/dotnet/runtime/blob/ea8262910cc1c072e612c6cdd0693e27ea218b92/src/libraries/System.Net.Primitives/src/System/Net/Cookie.cs#L430-L434

Since `localDomain` and `domain` are both `localhost` we're going into that branch, which doesn't happen when the machine/domain name is set to any other value.

The result is that instead of getting this cookie once in the tests I'm getting it _twice_ which fails the test:

```
Cookie:
    name = value1
    Domain: localhost
    Path: /path
    Port: "80 110,1050, 1090 ,1100"
    Secure: False
    When issued: 07/22/2020 17:46:32
    Expires: 01/01/0001 00:00:00 (expired? False)
    Don't save: False
    Comment:
    Uri for comments:
    Version: RFC 2109
```

It could also be a product bug

Reply:
thanks for explanation. This is not my primary are and I'll ask around.